pkg/wrapper/nmt_wrapper.go

```diff
@@ -25,19 +25,27 @@ type ErasuredNamespacedMerkleTree struct {
 	squareSize uint64 // note: this refers to the width of the original square before erasure-coded
 	options    []nmt.Option
 	tree       *nmt.NamespacedMerkleTree
-	idx        *rsmt2d.SquareIndex
+	// axisIndex is the index of the axis that this tree is on. This is passed
+	// by rsmt2d and used to help determine which quadrant each leaf belongs to.
+	axisIndex uint64
+	// shareIndex is the index of the share in a row or column that is being
+	// pushed to the tree. It is expected to be in the range: 0 <= shareIndex <
+	// 2*squareSize. shareIndex is used to help determine which quadrant each
+	// leaf belongs to.
+	shareIndex uint64
```

Comment:
It worth mentioning that the `shareIndex` also keeps track of the number of shares (i.e., the total number of leaves) that have been added to the tree so far.

Reply:
sure thing! https://github.com/celestiaorg/celestia-app/pull/1418/commits/afd0849b85ad97aa6694d6fd6ec9e46d57c4ebb8